File: benchmarks/CryptoBench/python-prompts/decode_message.py
from Crypto.Cipher import AES
from Crypto.Random import get_random_bytes

MODE = AES.MODE_EAX

# decode_message: Bytes Bytes -> Bytes
# Decodes the given message with the given key
def decode_message(key: bytes, encoded_message: bytes) -> bytes:
    # <solution>
    cipher = AES.new(key, mode=MODE)
    return cipher.decrypt(encoded_message)

# <tests>
def test_message(data: bytes):
    key = get_random_bytes(32)
    cipher = AES.new(key, mode=MODE)
    message = cipher.encrypt(data)
    assert decode_message(key, message), data

test_message(b"A good message")
test_message(b"A great day to live")
test_message(b"No time to die")

File: benchmarks/CryptoBench/lua-prompts/decode_message.lua
local cipher = require("openssl.cipher")
local rand = require("openssl.rand")

local aes_ecb = cipher.new("AES-256-ECB")

-- decode_message: String String -> String
-- Decodes the given message with the given key
local function decode_message(key, encoded_message)
    -- <solution>
    aes_ecb:decrypt(key)
    return aes_ecb:update(encoded_message) .. aes_ecb:final()
end

-- <tests>
local function test_message(data)
    local key = rand.bytes(32)

    aes_ecb:encrypt(key)
    local message = aes_ecb:update(data) .. aes_ecb:final()
    local decoded = decode_message(key, message)

    if decoded ~= data then
        error("Decoded message does not match original!")
    end
end

test_message("A good message")
test_message("A great day to live")
test_message("No time to die")
